packages/hash/api/src/graph/knowledge/system-types/block.ts

```diff
@@ -0,0 +1,210 @@
+    entityType,
+    actorId,
+  });
+
+  await createLinkEntity(ctx, {
+    linkEntityType: SYSTEM_TYPES.linkEntityType.blockData,
+    leftEntityId: entity.metadata.editionId.baseId,
+    rightEntityId: blockData.metadata.editionId.baseId,
+    ownedById,
+    actorId,
+  });
+
+  return getBlockFromEntity({ entity });
+};
+
+/**
+ * Get the linked block data entity of the block.
+ *
+ * @param params.block - the block
+ */
+export const getBlockData: ImpureGraphFunction<
+  { block: Block },
+  Promise<Entity>
+> = async (ctx, { block }) => {
+  const outgoingBlockDataLinks = await getEntityOutgoingLinks(ctx, {
+    entity: block.entity,
+    linkEntityType: SYSTEM_TYPES.linkEntityType.blockData,
+  });
+
+  const outgoingBlockDataLink = outgoingBlockDataLinks[0];
+
+  if (!outgoingBlockDataLink) {
+    throw new Error(
+      `Block with entityId ${block.entity.metadata.editionId.baseId} does not have an outgoing blockData link`,
+    );
+  }
+
+  return getLinkEntityRightEntity(ctx, { linkEntity: outgoingBlockDataLink });
+};
+
+/**
+ * Update the linked block data entity of a block.
+ *
+ * @param params.block - the block
+ * @param params.newBlockDataEntity - the new block data entity
+ * @param params.actorId - the id of the account that is updating the block data entity
+ */
+export const updateBlockDataEntity: ImpureGraphFunction<
+  {
+    block: Block;
+    newBlockDataEntity: Entity;
+    actorId: AccountId;
+  },
+  Promise<void>
+> = async (ctx, params) => {
+  const { block, newBlockDataEntity, actorId } = params;
+  const outgoingBlockDataLinks = await getEntityOutgoingLinks(ctx, {
+    entity: block.entity,
+    linkEntityType: SYSTEM_TYPES.linkEntityType.blockData,
+  });
+
+  const outgoingBlockDataLink = outgoingBlockDataLinks[0];
+
+  if (!outgoingBlockDataLink) {
+    throw new Error(
+      `Block with entityId ${block.entity.metadata.editionId.baseId} does not have an outgoing block data link`,
+    );
+  }
+
+  const existingBlockDataEntity = await getLinkEntityRightEntity(ctx, {
+    linkEntity: outgoingBlockDataLink,
+  });
+
+  if (
+    existingBlockDataEntity.metadata.editionId.baseId ===
+    newBlockDataEntity.metadata.editionId.baseId
+  ) {
+    throw new Error(
+      `The block with entity id ${existingBlockDataEntity.metadata.editionId.baseId} already has a linked block data entity with entity id ${newBlockDataEntity.metadata.editionId.baseId}`,
+    );
+  }
+
+  await archiveEntity(ctx, { entity: outgoingBlockDataLink, actorId });
+
+  await createLinkEntity(ctx, {
+    linkEntityType: SYSTEM_TYPES.linkEntityType.blockData,
+    leftEntityId: block.entity.metadata.editionId.baseId,
+    rightEntityId: newBlockDataEntity.metadata.editionId.baseId,
+    ownedById: extractOwnedByIdFromEntityId(
+      block.entity.metadata.editionId.baseId,
+    ),
+    actorId,
+  });
+};
+
+/**
+ * Get the comment of a block.
+ *
+ * @param params.block - the block
+ */
+export const getBlockComments: ImpureGraphFunction<
```

Comment:
It's a good question, but I think it would be inconsistent as this method gets the comments of a block, just like the `getPageBlocks` method (in the `page.ts` file) gets all the blocks of a page.

Reply:
Okay, that's fair. As long as we're consistent. Maybe if we make it clear that such methods should reside the place where their parameters are defined. So  since we're specifying a `Block` here, it should be in this file